runtime/interpreter/value.go

```diff
@@ -17762,6 +17763,7 @@ func (v *CompositeValue) forEachAttachment(interpreter *Interpreter, _ LocationR
 			// attachments is added that takes a `fun (&Attachment): Void` callback, the `f` provided here
 			// should convert the provided attachment value into a reference before passing it to the user
 			// callback
+			attachment.setBaseValue(interpreter, v)
```

Comment:
This seems to be the fix, setting the base value on the attachment before invoking the iteration callback.

Are the other changes in the PR also fixes, i.e. removing the last parameter of `setBaseValue` function and adjusting the call-sites, or is that just a simplification?

Reply:
Yeah, while doing this I noticed that every call to `setBaseValue` computed the `baseAuthorization` the same way, so I removed the argument and moved that computation into the body of the method.

Reply:
Nice!